cmd/crawl.go

```diff
@@ -80,6 +94,10 @@ The overall limit to nested archive size unarchived should be controlled by both
 	cmd.Flags().UintVar(&nestedArchiveMaxDepth, "nested-archive-max-depth", 0, `The maximum depth to recurse into nested archives. A max depth of 0 will open up an archive on the filesystem but not any nested archives.`)
 	cmd.Flags().IntVar(&directoriesCrawledPerSecond, "directories-per-second-rate-limit", 0, `The maximum number of directories to crawl per second. 0 for unlimited.`)
 	cmd.Flags().IntVar(&archivesCrawledPerSecond, "archives-per-second-rate-limit", 0, `The maximum number of archives to scan per second. 0 for unlimited.`)
+	cmd.Flags().BoolVar(&enableObfuscationDetection, "enable-obfuscation-detection", true, `Enable applying partial bytecode matching to Jars that appear to be obfuscated.`)
+	cmd.Flags().BoolVar(&enablePartialMatchingOnAllClasses, "enable-partial-matching-on-all-classes", false, `Enable partial bytecode matching to all class files found.`)
+	cmd.Flags().Uint32Var(&obfuscatedClassNameAverageLength, "maximum-average-obfuscated-class-name-length", 3, `The maximum average class name length for classes within a Jar to be considered obfuscated.`)
```

Comment:
Not sure if proper place for full documentation is flags or README, but just reading through the flag descriptions it's not completely clear to me how the obfuscation flags interact.

It seems like:

* If `enable-partial-matching-on-all-classes` is true, then bytecode detection is run on all classes
ELSE

* If `enable-obfuscation-detection` is false, then no bytecode detection is run.
ELSE

Bytecode detection is run on classes that are considered obfuscated. This determination is made based on the maximum length of the class name and/or (not clear if it is an AND or OR) the maximum length of the package name (since the default values here are "3", I'm guessing this means that in general obfuscated output is short, so that's the heuristic?)